lib/galaxy/webapps/galaxy/services/visualizations.py

```diff
@@ -32,3 +38,27 @@ def __init__(
         self.shareable_service = ShareableService(self.manager, self.serializer, notification_manager)
 
     # TODO: add the rest of the API actions here and call them directly from the API controller
+
+    def index(
+        self,
+        trans,
+        payload: VisualizationIndexQueryPayload,
+        include_total_count: bool = False,
+    ) -> Tuple[VisualizationSummaryList, int]:
+        """Return a list of Visualizations viewable by the user
+
+        :rtype:     list
+        :returns:   dictionaries containing Visualization details
+        """
+        if not trans.user_is_admin:
+            user_id = trans.user and trans.user.id
+            if payload.user_id and payload.user_id != user_id:
+                raise exceptions.AdminRequiredException("Only admins can index the visualizations of others")
+
+        entries, total_matches = self.manager.index_query(trans, payload, include_total_count)
+        return (
+            VisualizationSummaryList.construct(
+                __root__=[trans.security.encode_all_ids(entry.to_dict(), recursive=True) for entry in entries]
```

Comment:
```suggestion
            ```suggestion
                VisualizationSummaryList(__root__=[entry.to_dict() for entry in entries])
```
the id handling should be determined by the types in your response model, and not bypassing the validation will then make sure you return the correct field types.
```